backend/src/main/java/com/my/shope/backend/product/ProductService.java

```diff
@@ -38,7 +39,7 @@ public Product getProductById(String id) {
     }
 
     public Product updateProduct(Product theProduct) {
-
+       theProduct.setId(null);
```

Comment:
Wenn du ein bestimmtes Produkt updaten willst, dann darfst du hier die ID nicht resetten, weil das Repo sonst ein neues Produkt anlegt. 
Oder hast hier einen bestimmten Grund, dass die Produkte mehrmals in der DB unter unterschiedlichen IDs liegen sollen?

Reply:
hast du abslut recht ich habe das bei updateOrder gemerkt und ich ich habe vergessen da auch zu updaten aber eben habe ich gemacht danke:)